src/Spice86/ViewModels/DebuggerLineViewModel.cs

```diff
@@ -0,0 +1,162 @@
+using Spice86.Core.Emulator.Function;
+using Spice86.Core.Emulator.VM.Breakpoint;
+using Spice86.Models.Debugging;
+using Spice86.Shared.Emulator.Memory;
+
+using System.Text.Json;
+
+/// <summary>
+///     View model for a line in the debugger.
+/// </summary>
+public partial class DebuggerLineViewModel : ViewModelBase {
+    private readonly BreakpointsViewModel? _breakpointsViewModel;
+    private readonly State _cpuState;
+
+    private readonly Formatter _formatter = new MasmFormatter(new FormatterOptions {
+        AddLeadingZeroToHexNumbers = false,
+        AlwaysShowSegmentRegister = true,
+        HexPrefix = "0x",
+        MasmSymbolDisplInBrackets = false
+    });
+
+    private readonly Instruction _info;
+
+    [ObservableProperty]
+    private BreakpointViewModel? _breakpoint;
+
+    [ObservableProperty]
+    private bool _isCurrentInstruction;
+
+    [ObservableProperty]
+    private bool _isSelected;
+
+    [ObservableProperty]
+    private bool? _willJump;
+
+    public DebuggerLineViewModel(EnrichedInstruction instruction, State cpuState, BreakpointsViewModel? breakpointsViewModel = null) {
+        _info = instruction.Instruction;
+        _cpuState = cpuState;
+        _breakpointsViewModel = breakpointsViewModel;
+        ByteString = string.Join(' ', instruction.Bytes.Select(b => b.ToString("X2")));
+        Function = instruction.Function;
+        SegmentedAddress = instruction.SegmentedAddress;
+        Address = SegmentedAddress.Linear;
+        NextAddress = (uint)(Address + _info.Length);
+
+        // We expect there to be at most 1 execution breakpoint per line, so we use SingleOrDefault.
+        Breakpoint = instruction.Breakpoints.SingleOrDefault(breakpoint => breakpoint.Type == BreakPointType.CPU_EXECUTION_ADDRESS);
+
+        // Generate the formatted disassembly text
+        GenerateFormattedDisassembly();
+    }
+
+    public string ByteString { get; }
+    public FunctionInformation? Function { get; }
+    public SegmentedAddress SegmentedAddress { get; }
+
+    /// <summary>
+    ///     The physical address of this instruction in memory.
+    /// </summary>
+    public uint Address { get; }
+
+    public bool ContinuesToNextInstruction => _info.FlowControl == FlowControl.Next;
+    public bool CanBeSteppedOver => _info.FlowControl is FlowControl.Call or FlowControl.IndirectCall or FlowControl.Interrupt;
+    public uint NextAddress { get; private set; }
+
+    public string Disassembly => _info.ToString();
+
+    /// <summary>
+    ///     Gets a collection of formatted text segments for the disassembly with syntax highlighting.
+    /// </summary>
+    public List<FormattedTextSegment> DisassemblySegments { get; private set; } = [];
+
+    public void ApplyCpuState() {
+        // Initialize WillJump to null for non-conditional branches
+        WillJump = null;
+
+        if (_info.FlowControl == FlowControl.ConditionalBranch) {
+            WillJump = _info.ConditionCode switch {
```

Comment:
Tried to run it in local but for loop it always shows not taken because ConditionCode is None.
Still does the job for other conditional jumps

Reply:
I removed the WIP jump indicator. That was not yet ready for this phase1.